staging/src/k8s.io/client-go/tools/cache/shared_informer_test.go

```diff
@@ -18,33 +18,58 @@ package cache
 
 import (
 	"fmt"
+	"math"
 	"sync"
 	"testing"
 	"time"
 
 	"k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/clock"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/wait"
 	fcache "k8s.io/client-go/tools/cache/testing"
 )
 
+const timeFmt = "15:04:05.999999999"
+
 type testListener struct {
-	lock              sync.RWMutex
-	resyncPeriod      time.Duration
-	expectedItemNames sets.String
-	receivedItemNames []string
-	name              string
+	t                *testing.T
+	name             string
+	clock            clock.PassiveClock
+	resyncPeriod     time.Duration
+	checkTiming      bool // whether handle() checks timing
+	expectedItemKeys sets.String
+	lock             sync.RWMutex
+	whenReceived     map[string]time.Time // maps key to when last received
+}
+
+func (l *testListener) receivedItemKeys() sets.String {
+	l.lock.Lock()
+	defer l.lock.Unlock()
+	return sets.StringKeySet(l.whenReceived)
+}
+
+func (l *testListener) clearMemory() {
```

Comment:
@MikeSpreitzer can we name this `reinit()` or something, since it's really just clearing values that could interfere with its reuse?

Reply:
"reset" is the most traditional name for this sort of thing.

Reply:
"init" is short for "initialize", which often concerns all the fields of an object rather than just a few.
"reset" is not incorrect, but is sometimes used with a more general meaning.
I tried for a name that was more clearly indicative of the limited function involved.

Reply:
I decided to go with "reset".